### ISSUE TITLE
[no-explicit-any] `ignoreRestArgs` doesn't detect interface `any` correctly

### ISSUE THREAD

**Issue by Dilatorily** [typescript-eslint/typescript-eslint#776]
**Repro**

```JSON
{
  "rules": {
    "typescript/no-explicit-any": ["error", { "ignoreRestArgs": true }]
  }
}
```

```TS
interface Callback {
  (...args: any[]): void;
}
```

**Expected Result**
No errors raised

**Actual Result**
ESLint raises an error in the interface

**Additional Info**
N/A

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.0` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.4.0` |
| `npm`                              | `6.9.0` |

**Comment by briman0094** [comment]
This still occurs on `2.0.0` with constructor interfaces, e.g. `new ( ...args: any[] ) => MyType`.

**Comment by bradzacher** [comment]
Please file a new issue for this @briman0094, and we can investigate.
